### DISCUSSION TITLE
Non-tweakable version caching of a language segment?

### DISCUSSION THREAD

**Discussion by lewis-yeung** [JanDeDobbeleer/oh-my-posh#5831] [General]
In v24, the version caching of a language segment cannot be tweaked by the `cache` settings. Instead, it always fetches the global cache (in omp.cache) which persists for one week.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung yes, because in my experience I didn't see that change often. As doing that call once a week across all shell instances felt like it was sufficient in combination with the new cache functionality anyways. Unless that's redundant and we can simply resort to the cache alone.

**Reply by lewis-yeung** [reply]
That causes problems indeed, e.g., detecting the Python version in a virtual environment, and some other cases where we'd like to switch between different versions of a language tool.

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung that is a fair callout, and contextual invalidation isn't the way to go. That said, you do still invalidate the cache manually in those cases too, or not? Because even with a tweak-able parameter, it can happen between that interval?

**Reply by lewis-yeung** [reply]
@JanDeDobbeleer To be precise, for these cases the cache needs to be disabled, however, this is impossible now since the global cache is always picked up forcibly.

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung ok, thanks. Makes sense. I'll adjust this.

**Reply by lewis-yeung** [reply]
@JanDeDobbeleer Tested just now, while #5859 didn't fix the issue. Even with the `cache_duration` set to `none`, the version data still appears in the global cache.

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung did you clear the cache after changing that value?

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung tested again on my end, works as designed. Did forget to adjust the schema again though 😅

**Reply by lewis-yeung** [reply]
> did you clear the cache after changing that value?

@JanDeDobbeleer Yes, and restarted the shell.

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung are you sure you're on main? Because this one I really can't reproduce. Unless I didn't commit correctly...

EDIT: checked the code, that checks out. Where did you set the `cache_duration`?

**Reply by lewis-yeung** [reply]
@JanDeDobbeleer Yes, exactly.

**Reply by lewis-yeung** [reply]
Part of the config (a `go` segment):

```yaml
- type: go
  style: plain
  properties:
    home_enabled: true
    display_mode: always
  cache_duration: none
  template: '{{if .Full}} {{.Full}}{{end}}'
```

**Reply by lewis-yeung** [reply]
It always writes a cache `version_golang` with the TTL of `604800`, into omp.cache.

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung `cache_duration` needs to be under properties 😅

**Reply by lewis-yeung** [reply]
@JanDeDobbeleer A stupid mistake everyday (1/1). 🤦‍ Forgive me...

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung I've been breaking stuff for the past week, I'm the last person to judge anyone on this. Your input and contributions are amazingly valuable, you'll be fine 👍🏻

**Reply by lewis-yeung** [reply]
@JanDeDobbeleer Thank you. It's always tough to move into a new phase. 👍

Happy to close this now.
